graylog2-server/src/main/java/org/graylog2/indexer/retention/strategies/AbstractIndexRetentionStrategy.java

```diff
@@ -70,14 +72,15 @@ private void retainTimeBased(IndexSet indexSet, TimeBasedSizeOptimizingStrategyC
         final Map<String, Set<String>> deflectorIndices = indexSet.getAllIndexAliases();
 
         // Account for DST and time zones in determining age
-        final long cutoff = Instant.now().minus(smartConfig.indexLifetimeSoft()).toEpochMilli();
+        final long cutoff = clock.instantNow().minus(smartConfig.indexLifetimeSoft()).toEpochMilli();
         final int removeCount = (int)deflectorIndices.keySet()
                 .stream()
                 .filter(indexName -> !indices.isReopened(indexName))
+                .filter(indexName -> !hasCurrentWriteAlias(indexSet, deflectorIndices, indexName))
                 .filter(indexName -> {
                     DateTime closingDate = indices.indexClosingDate(indexName)
                             .orElseThrow(() -> new IllegalStateException(f("Index %s has no closing date - retention failed", indexName)));
-                    return closingDate.isBefore(cutoff);
+                    return closingDate.isBefore(cutoff + 1);
```

Comment:
1 ms change - what difference does that make when retention periods are measured in days?

Reply:
It make the comparison inclusive instead of exclusive. doesn't matter that much, but it felt nicer while testing it.